src/Spago/Purs.hs

```diff
@@ -74,24 +74,50 @@ repl sourcePaths extraArgs = do
 
   viewShell $ callCommand $ Text.unpack cmd
 
-
-bundle :: HasLogFunc env => WithMain -> WithSrcMap -> ModuleName -> TargetPath -> RIO env ()
-bundle withMain withSourceMap (ModuleName moduleName) (TargetPath targetPath) = do
-  let main = case withMain of
-        WithMain    -> " --main " <> moduleName
-        WithoutMain -> ""
-
-      sourceMap = case withSourceMap of 
-        WithSrcMap    -> " --source-maps"
-        WithoutSrcMap -> ""
-
-      cmd
-        = "purs bundle \"output/*/*.js\""
-        <> " -m " <> moduleName
-        <> main
-        <> " -o " <> targetPath
-        <> sourceMap
-
+getESBuild :: HasLogFunc env => RIO env Text
+getESBuild = do
+  maybeESBuild <- findExecutable "esbuild"
+  case maybeESBuild of
+    Nothing -> die [ "Failed to find esbuild. See https://esbuild.github.io/getting-started/#install-esbuild for ways to install esbuild." ]
+    Just esBuild -> pure esBuild
+
+bundle :: HasLogFunc env => ModuleSystem -> WithMain -> WithSrcMap -> ModuleName -> TargetPath -> Platform -> Minify -> RIO env ()
+bundle ESM withMain _ (ModuleName moduleName) (TargetPath targetPath) platform minify = do
+  esbuild <- getESBuild
+  let 
+    platformOpt = case platform of 
+      Browser -> "browser"
+      Node -> "node"
+    minifyOpt = case minify of 
+      NoMinify -> ""
+      Minify -> " --minify"
+    cmd = case withMain of
+      WithMain -> 
+        "echo \"import { main } from './output/" <> moduleName <> "/index.js'\nmain()\" | "
+        <> esbuild <> " --platform=" <> platformOpt <> minifyOpt <> " --bundle "
+        <> " --outfile=" <> targetPath
+      WithoutMain -> 
+        esbuild <> " --platform=" <> platformOpt <> minifyOpt <> " --bundle " 
```

Comment:
I'm not sure all this logic about `esbuild` should go here, since this module is supposed to only be dealing with `purs` itself. Would it be possible to pull this logic up in the `Build` module?

Reply:
yeah I was also wondering where best to put that, and then decided to have it next to the purs bundle so to not have it in two different places, but I can move it to `Build.hs`